features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/RoomListPresenter.kt

```diff
@@ -137,7 +141,11 @@ class RoomListPresenter @Inject constructor(
             displayVerificationPrompt = displayVerificationPrompt,
             snackbarMessage = snackbarMessage,
             hasNetworkConnection = networkConnectionStatus == NetworkStatus.Online,
-            displayInvites = invites.isNotEmpty(),
+            invitesState = when {
+                invites.isEmpty() -> InvitesState.NoInvites
+                invites.map { it.identifier() }.all(seenInvites::contains) -> InvitesState.SeenInvites
```

Comment:
This will be done on the mainthread, not ideal

Reply:
Oh, yeah, ick. Good call. I've made a separate data source for the invite state, which both stops it being recomputed on the main thread constantly and encapsulates all the logic nicely.